radis/reports/api/viewsets.py

```
@@ -264,185 +371,246 @@ def on_commit():
+            lambda: serializer.data, thread_sensitive=True
+        )()
+        http_status = (
+            status.HTTP_201_CREATED if report is None else status.HTTP_200_OK
+        )
+        return Response(response_data, status=http_status)
+
+    async def adestroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        try:
+            report = await Report.objects.aget(document_id=kwargs[self.lookup_field])
+        except Report.DoesNotExist:
+            raise Http404
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)